### PR TITLE
Large code/documentation cleanup

### PR DESCRIPTION
**Pull request by ghost**
## Summary

This PR does a ton of things, but I'll attempt to summarize here:
- Fixes a large amount of spelling and grammatical errors in code (not entirely necessary, but it removes linting typo/proofreading errors).
- Fixes broken links on documentation ~~(not all fixed, see below for remaining)~~.
- Polishes redundant code -- a few examples:
  - Remove `_Undefined` in favor of `MISSING` in `abc.py`.
  - Remove unused imports in several files.
  - Edit `tuple` returns to return `val, val, ...` instead of `(val, val, ...)`.
- Shortens lines that extend past L120.
- Fixes some small bugs.
- More small things I am probably glossing over.

~~Regarding broken links, there are still a few that remain and I have yet to get around to fixing yet. Namely:~~
- ~~Auto mod objects (i.e. `AutoModRule`, `AutoModActionExecutionEvent`, and more) that are referenced in the documentation ([here](https://docs.pycord.dev/en/master/api.html#discord.on_auto_moderation_rule_create) and [here](https://docs.pycord.dev/en/master/api.html#discord.Guild.fetch_auto_moderation_rules) from what I've found) are broken links as the objects are not currently listed anywhere in the documentation.~~ (will be handled in a separate PR)
- ~~Any references to `ForumChannel` in the documentation are broken as it is not currently listed anywhere. (found [here](https://docs.pycord.dev/en/master/api.html#discord.abc.GuildChannel), [here](https://docs.pycord.dev/en/master/api.html#discord.Guild.forum_channels), [here](https://docs.pycord.dev/en/master/api.html#discord.Guild.create_forum_channel), [here](https://docs.pycord.dev/en/master/api.html#discord.CategoryChannel.forum_channels), and [here](https://docs.pycord.dev/en/master/api.html#discord.CategoryChannel.create_forum_channel) so far.)~~
- ~~Not all members of `TextChannel` are being displayed on the docs, resulting in missing attributes on the object itself and a broken reference. (broken reference found [here](https://docs.pycord.dev/en/master/api.html#discord.AuditLogDiff.slowmode_delay), haven't found any others so far.)~~

~~Also, lastly, this PR will probably remain in draft until the broken links listed above are fixed (any help is much appreciated, I can't do much more documentation work for my own sanity) and the changes made here are ensured to not cause any issues (once again, any testing done is much appreciated).~~

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by Lulalaby** [comment]
Please don't say this should go into release too 😅

**Review by Lulalaby — DISMISSED** [review]
tbh LGTM so far, @Dorukyum what you think

File: discord/bot.py [hunk 1]
```diff
@@ -1374,7 +1373,7 @@ class Bot(BotBase, Client):
     anything that you can do with a :class:`discord.Client` you can do with
     this bot.
 
-    This class also subclasses :class:`.ApplicationCommandMixin` to provide the functionality
+    This class also subclasses ``ApplicationCommandMixin`` to provide the functionality
```
Comment (Dorukyum) [inline discord/bot.py:1376]: What is the reason behind this?

File: discord/channel.py [hunk 2]
```diff
@@ -1821,9 +1821,8 @@ async def edit(self, *, reason=None, **options):
             category.
         reason: Optional[:class:`str`]
             The reason for editing this channel. Shows up on the audit log.
-        overwrites: :class:`Mapping`
-            A :class:`Mapping` of target (either a role or a member) to
-            :class:`PermissionOverwrite` to apply to the channel.
+        overwrites: Dict[Union[:class:`Role`, :class:`Member`, :class:`Snowflake`], :class:`PermissionOverwrite`]
+            The overwrites to apply to the channel. Useful for creating secret channels.
```
Comment (Dorukyum) [inline discord/channel.py:1825]:
```suggestion
            The overwrites to apply to channel permissions. Useful for creating secret channels.
```

File: discord/channel.py [hunk 3]
```diff
@@ -1986,9 +1985,8 @@ async def edit(self, *, reason=None, **options):
             To mark the category as NSFW or not.
         reason: Optional[:class:`str`]
             The reason for editing this category. Shows up on the audit log.
-        overwrites: :class:`Mapping`
-            A :class:`Mapping` of target (either a role or a member) to
-            :class:`PermissionOverwrite` to apply to the channel.
+        overwrites: Dict[Union[:class:`Role`, :class:`Member`, :class:`Snowflake`], :class:`PermissionOverwrite`]
+            The overwrites to apply to the channel. Useful for creating secret channels.
```
Comment (Dorukyum) [inline discord/channel.py:1989]:
```suggestion
            The overwrites to apply to channel permissions. Useful for creating secret channels.
```

File: discord/ext/commands/core.py [hunk 4]
```diff
@@ -1606,8 +1607,8 @@ def command(
 ) -> Callable[
     [
         Union[
-            Callable[Concatenate[CogT, ContextT, P]], Coro[T],
-            Callable[Concatenate[ContextT, P]], Coro[T],
+            Callable[[Concatenate[CogT, ContextT, P]], Coro[T]],
+            Callable[[Concatenate[ContextT, P]], Coro[T]],
```
Comment (Dorukyum) [inline discord/ext/commands/core.py:1611]: These brakcets are incorrect for python 3.10+, which is why I added this overload.

File: discord/ext/commands/view.py [hunk 5]
```diff
@@ -87,20 +87,20 @@ def skip_ws(self):
 
     def skip_string(self, string):
         strlen = len(string)
-        if self.buffer[self.index : self.index + strlen] == string:
+        if self.buffer[self.index: self.index + strlen] == string:
```
Comment (Dorukyum) [inline discord/ext/commands/view.py:90]: Shouldn't these spaces be there according to pep8

File: discord/http.py [hunk 6]
```diff
@@ -2494,7 +2494,7 @@ def create_followup_message(
         self,
         application_id: Snowflake,
         token: str,
-        files: List[File] = [],
+        files: List[File] = None,
```
Comment (Dorukyum) [inline discord/http.py:2497]:
```suggestion
        files: Optional[List[File]] = None,
```

**Review by Dorukyum — DISMISSED** [review]

**Comment by EmreTech** [comment]
> Shortens lines that extend past L120.

Personally I think instead that another Black linting PR (that re-adds the CI with some differences) would be better for this instead of doing it in this PR.

File: discord/bot.py (hunk 1 above)
Comment (baronkobama) [reply discord/bot.py:1376]: It's a broken link on the docs because `ApplicationCommandMixin` isn't documented anywhere, so I converted it to a normal code block.

File: discord/ext/commands/core.py (hunk 4 above)
Comment (baronkobama) [reply discord/ext/commands/core.py:1611]: Okay, if there's a purpose to it then I will definitely revert, just saw that it was raising errors in my linter.

File: discord/ext/commands/view.py (hunk 5 above)
Comment (baronkobama) [reply discord/ext/commands/view.py:90]: It's not entirely clear as `pycodestyle` raises a weak warning with that but it does appear pep8 wants it that way. Will revert these.

File: discord/activity.py [hunk 7]
```diff
@@ -509,7 +508,7 @@ def twitch_name(self):
         """Optional[:class:`str`]: If provided, the twitch name of the user streaming.
 
         This corresponds to the ``large_image`` key of the :attr:`Streaming.assets`
-        dictionary if it starts with ``twitch:``. Typically set by the Discord client.
+        dictionary if it starts with ``twitch:``. Typically, set by the Discord client.
```
Comment (plun1331) [inline discord/activity.py:511]: Makes more grammatical sense without the comma imo

File: discord/cog.py [hunk 8]
```diff
@@ -415,8 +415,11 @@ def bot_check_once(self, ctx: ApplicationContext) -> bool:
         """A special method that registers as a :meth:`.Bot.check_once`
         check.
 
-        This function **can** be a coroutine and must take a sole parameter,
-        ``ctx``, to represent the :class:`.Context`.
+        This function **can** be a coroutine.
+
+        Parameters
+        -----------
```
Comment (plun1331) [inline discord/cog.py:421]: Parameter name is missing

File: discord/ext/pages/pagination.py [hunk 9]
```diff
@@ -220,12 +220,14 @@ class PageGroup:
 
     .. note::
 
-        If multiple ``PageGroup`` objects have different options, they should all be set explicitly when creating each instance.
+        If multiple ``PageGroup`` objects have different options,
```
Comment (plun1331) [inline discord/ext/pages/pagination.py:223]: This could reference the object instead of being code

File: discord/guild.py [hunk 10]
```diff
@@ -226,17 +227,21 @@ class Guild(Hashable):
...
-        - ``SEVEN_DAY_THREAD_ARCHIVE``: Guild has access to the seven day archive time for threads.
-        - ``TEXT_IN_VOICE_ENABLED``: Guild has a chat button inside voice channels that opens a dedicated text channel in a sidebar similar to thread view.
-        - ``THREAD_DEFAULT_AUTO_ARCHIVE_DURATION``: Unknown, presumably used for testing changes to the thread default auto archive duration..
-        - ``THREADS_ENABLED_TESTING``: Used by bot developers to test their bots with threads in guilds with 5 or less members and a bot. Also gives the premium thread features.
-        - ``THREE_DAY_THREAD_ARCHIVE``: Guild has access to the three day archive time for threads.
+        - ``SEVEN_DAY_THREAD_ARCHIVE``: Guild has access to the seven-day archive time for threads.
+        - ``TEXT_IN_VOICE_ENABLED``: Guild has a chat button inside voice channels that opens a dedicated text channel
+                                     in a sidebar similar to thread view.
+        - ``THREAD_DEFAULT_AUTO_ARCHIVE_DURATION``: Unknown, presumably used for testing changes to the thread default
+                                                    auto archive duration...
```
Comment (plun1331) [inline discord/guild.py:241]: I think it's more likely that this was meant to be a single period, not an ellipsis

File: discord/guild.py [hunk 11]
```diff
@@ -464,7 +469,7 @@ def _add_role(self, role: Role, /) -> None:
         self._roles[role.id] = role
 
     def _remove_role(self, role_id: int, /) -> Role:
-        # this raises KeyError if it fails..
+        # this raises KeyError if it fails...
```
Comment (plun1331) [inline discord/guild.py:472]: I think it's more likely that this was meant to be a single period, not an ellipsis

File: discord/abc.py [hunk 12]
```diff
@@ -658,7 +650,7 @@ def permissions_for(self, obj: Union[Member, Role], /) -> Permissions:
         # (or otherwise) are then OR'd together.
         # After the role permissions are resolved, the member permissions
         # have to take into effect.
-        # After all that is done.. you have to do the following:
+        # After all that is done... you have to do the following:
```
Comment (plun1331) [inline discord/abc.py:653]:
```suggestion
        # After all that is done, you have to do the following:
```

File: discord/mentions.py [hunk 13]
```diff
@@ -118,14 +118,14 @@ def to_dict(self) -> AllowedMentionsPayload:
         if self.everyone:
             parse.append("everyone")
 
-        if self.users == True:
+        if self.users:
             parse.append("users")
-        elif self.users != False:
+        elif not self.users:
```
Comment (plun1331) [inline discord/mentions.py:123]: This seems to have been like this to allow for the passing of a list of users, so it should probably remain close to the same, e.g.
```py
if self.users is True:
    parse.append("users")
elif self.users is not False:
```
*Github won't allow suggestions on this because of the deleted line*

File: discord/mentions.py [hunk 14]
```diff
@@ -118,14 +118,14 @@ def to_dict(self) -> AllowedMentionsPayload:
...
             data["users"] = [x.id for x in self.users]
 
-        if self.roles == True:
+        if self.roles:
             parse.append("roles")
-        elif self.roles != False:
+        elif not self.roles:
```
Comment (plun1331) [inline discord/mentions.py:128]: Same as above

File: discord/stage_instance.py [hunk 15]
```diff
@@ -101,10 +101,8 @@ def _update(self, data: StageInstancePayload):
         self.topic: str = data["topic"]
         self.privacy_level: StagePrivacyLevel = try_enum(StagePrivacyLevel, data["privacy_level"])
         self.discoverable_disabled: bool = data.get("discoverable_disabled", False)
-        event_id = data.get("guild_scheduled_event")
-        if event_id is not None:
-            event_id = int(event_id)
-        self.scheduled_event = self.guild.get_scheduled_event(event_id)
+        if (event_id := data.get("guild_scheduled_event_id", None)) is not None:
+            self.scheduled_event = self.guild.get_scheduled_event(int(event_id))
```
Comment (plun1331) [inline discord/stage_instance.py:105]: Scheduled event should be sent to none if there is no event id

File: discord/state.py [hunk 16]
```diff
@@ -1789,7 +1789,7 @@ async def _delay_ready(self) -> None:
         except AttributeError:
             pass  # already been deleted somehow
 
-        # regular users cannot shard so we won't worry about it here.
+        # regular users cannot shard, so we won't worry about it here.
```
Comment (plun1331) [inline discord/state.py:1792]: Comment probably isn't needed anymore, we don't support user bots

File: discord/utils.py [hunk 17]
```diff
@@ -1078,8 +1078,8 @@ def format_dt(dt: datetime.datetime, /, style: Optional[TimestampStyle] = None)
 
 
 def generate_snowflake(dt: Optional[datetime.datetime] = None) -> int:
-    """Returns a numeric snowflake pretending to be created at the given date but more accurate and random than time_snowflake.
-    If dt is not passed, it makes one from the current time using utcnow.
+    """Returns a numeric snowflake pretending to be created at the given date but more accurate and random
+    than time_snowflake. If dt is not passed, it makes one from the current time using utcnow.
```
Comment (plun1331) [inline discord/utils.py:1082]: Link to the function

**Comment by baronkobama** [comment]
> Personally I think instead that another Black linting PR (that re-adds the CI with some differences) would be better for this instead of doing it in this PR.

Should this be done? I don't see a reason to undo those changes just so it can be done again, but if others think so then I can revert them.

File: discord/bot.py (hunk 1 above)
Comment (Lulalaby) [reply discord/bot.py:1376]: Can we document it instead

File: discord/guild.py [hunk 18]
```diff
@@ -214,7 +214,8 @@ class Guild(Hashable):
         - ``DISCOVERABLE``: Guild shows up in Server Discovery.
         - ``FEATURABLE``: Guild is able to be featured in Server Discovery.
```
Comment (Lulalaby) [inline discord/guild.py:215]: Isn't used anymore

File: discord/guild.py [hunk 19]
```diff
@@ -226,17 +227,21 @@ class Guild(Hashable):
...
-        - ``THREADS_ENABLED_TESTING``: Used by bot developers to test their bots with threads in guilds with 5 or less members and a bot. Also gives the premium thread features.
-        - ``THREE_DAY_THREAD_ARCHIVE``: Guild has access to the three day archive time for threads.
+        - ``SEVEN_DAY_THREAD_ARCHIVE``: Guild has access to the seven-day archive time for threads.
+        - ``TEXT_IN_VOICE_ENABLED``: Guild has a chat button inside voice channels that opens a dedicated text channel
+                                     in a sidebar similar to thread view.
+        - ``THREAD_DEFAULT_AUTO_ARCHIVE_DURATION``: Unknown, presumably used for testing changes to the thread default
+                                                    auto archive duration.
+        - ``THREADS_ENABLED_TESTING``: Used by bot developers to test their bots with threads in guilds with 5 or fewer
+                                       members and a bot. Also gives the premium thread features.
+        - ``THREE_DAY_THREAD_ARCHIVE``: Guild has access to the three-day archive time for threads.
```
Comment (Lulalaby) [inline discord/guild.py:244]:
```suggestion
        - ``TEXT_IN_VOICE_ENABLED``: Guild has a chat button inside voice channels that opens a dedicated text channel
                                     in a sidebar similar to thread view.
        - ``THREADS_ENABLED_TESTING``: Used by bot developers to test their bots with threads in guilds with 5 or fewer
                                       members and a bot. Also gives the premium thread features.
```

Obsolete

**Review by Lulalaby — CHANGES_REQUESTED** [review]

File: discord/permissions.py [hunk 20]
```diff
@@ -46,6 +46,7 @@
     "PermissionOverwrite",
 )
 
+
```
[the remark this answers is not in the document]
Comment (baronkobama) [reply discord/permissions.py:49]: There needs to be 2 empty lines between a class definition and any lines before or after.

[the remark this answers is not in the document]
Comment (Lulalaby) [reply discord/permissions.py:49]: Ohhh Python stuff I guess 😂

**Comment by Lulalaby** [comment]
I meant only featureable ^^
GitHub mobile sucks 

I read your whole pr in bed because I couldn't sleep XD

**Comment by baronkobama** [comment]
> I meant only featureable ^^ GitHub mobile sucks

Oh lol, I'll undo that.

**Review by Lulalaby — DISMISSED** [review]
Within my sleepy state I declare the changes as LGTM

**Review by Lulalaby — DISMISSED** [review]
Again I'm sleepy and once more I approve this pr as LGTM.
_And because I only fixed a merge conflict_ YEEEEEEEEET

File: discord/commands/permissions.py [hunk 21]
```diff
@@ -48,7 +48,7 @@ def default_permissions(**perms: bool) -> Callable:
 
     Parameters
     ------------
-    perms
+    perms: :class:`bool`
```
Comment (Middledot) [inline discord/commands/permissions.py:51]:
```suggestion
    **perms: Dict[:class:`str`, :class:`bool`]
```

File: discord/http.py [hunk 22]
```diff
@@ -523,22 +522,23 @@ def send_multipart_helper(
...
-            )
-            form.append(
-                {
-                    "name": f"files[{index}]",
-                    "value": file.fp,
-                    "filename": file.filename,
-                    "content_type": "application/octet-stream",
-                }
-            )
+        if files:
```
Comment (Middledot) [inline discord/http.py:525]: Files is already required
https://github.com/Pycord-Development/pycord/blob/8e941f31e276eb5c46a0df4e41ab512e59023002/discord/http.py#L493

File: discord/scheduled_events.py [hunk 23]
```diff
@@ -200,8 +200,7 @@ def __init__(
         self.description: Optional[str] = data.get("description", None)
         self._cover: Optional[str] = data.get("image", None)
         self.start_time: datetime.datetime = datetime.datetime.fromisoformat(data.get("scheduled_start_time"))
-        end_time = data.get("scheduled_end_time", None)
-        if end_time != None:
+        if (end_time := data.get("scheduled_end_time", None)) is not None:
```
Comment (Middledot) [inline discord/scheduled_events.py:203]:
```suggestion
        if end_time := data.get("scheduled_end_time", None):
```

File: discord/stage_instance.py [hunk 24]
```diff
@@ -101,10 +101,11 @@ def _update(self, data: StageInstancePayload):
...
         self.discoverable_disabled: bool = data.get("discoverable_disabled", False)
-        event_id = data.get("guild_scheduled_event")
-        if event_id is not None:
-            event_id = int(event_id)
-        self.scheduled_event = self.guild.get_scheduled_event(event_id)
+
+        if (event_id := data.get("guild_scheduled_event_id", None)) is not None:
+            self.scheduled_event = self.guild.get_scheduled_event(int(event_id))
+        else:
+            self.scheduled_event = None
```
Comment (Middledot) [inline discord/stage_instance.py:108]:
```suggestion

        self.scheduled_event = None
        if event_id := data.get("guild_scheduled_event_id", None):
            self.scheduled_event = self.guild.get_scheduled_event(int(event_id))
```

File: docs/api.rst [hunk 25]
```diff
@@ -2812,9 +2814,9 @@ of :class:`enum.Enum`.
         - :attr:`~AuditLogDiff.description`
         - :attr:`~AuditLogDiff.channel`
         - :attr:`~AuditLogDiff.privacy_level`
-        - :attr:`~AuditLogDiff.location`
-        - :attr:`~AuditLogDiff.status`
-        - :attr:`~AuditLogDiff.location_type`
+        - :attr:`~discord.ScheduledEvent.location`
+        - :attr:`~discord.ScheduledEvent.status`
+        - :attr:`~discord.ScheduledEventLocation.type`
```
Comment (Middledot) [inline docs/api.rst:2819]: Why was this changed?

**Review by Middledot — COMMENTED** [review]
Otherwise lgtm

File: docs/api.rst (hunk 25 above)
Comment (baronkobama) [reply docs/api.rst:2819]: These were broken references due to AuditLogDiff not having those attributes.

File: discord/http.py (hunk 22 above)
Comment (baronkobama) [reply discord/http.py:525]: Good catch, not sure how I missed that lol

File: docs/api.rst (hunk 25 above)
Comment (Middledot) [reply docs/api.rst:2819]: Oh ok, I guess they could be documented in another PR

**Review by Middledot — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
